src/Stoa.ts

```diff
@@ -23,14 +24,85 @@ class Stoa {
             }
         });
 
+        /**
+         * When a request is received through the `/validators` handler
+         * Validators based on the block height if there is a height.
+         * If height null the most up to date state is expected.
+         */
         this.stoa.get("/validators",
             (req: express.Request, res: express.Response, next: express.NextFunction) => {
-            res.send(req.query.height);
+
+                var height: number = Number(req.query.height);
+
+                this.ledger_storage.getValidatorsAPI(height, null, (err: Error | null, rows: any[]) =>
+                {
+                    if (err != null)
+                    {
+                        console.error("Failed to data lookup to the DB: " + err);
+                        res.status(500).send("Failed to data lookup");
+                        return;
+                    }
+
+                    if (rows.length)
+                    {
+                        let out_put:Array<ValidatorData> = new Array<ValidatorData>();
+
+                        for (const row of rows)
+                        {
+                            let preimage: IPreimage = {distance: row.distance, hash: '0'} as IPreimage;
+                            var validator: ValidatorData | null =
+                                new ValidatorData(row.address, row.enrolled_at, row.stake, preimage);
+                            out_put.push(validator);
+                        }
+                        res.status(200).send(JSON.stringify(out_put));
+                    }
+                    else
+                    {
+                        res.status(200).send("no data");
+                    }
+                });
         });
 
+        /**
```

Comment:
Is this documentation javascript-style convention? Because it's written in a very weird way. Where are the verbs describing the actions?

When.. => Called when..
Validators based on => Returns a set of Validators based on..
If height null the most up to date state is expected. => If height was not provided the latest validator set is returned
If an address exists => If an address was provided, return the validator data of the address if it exists

Reply:
Thank you.

Reply:
I applied JSDoc style.